components/brave_shields/core/common/ad_block_only_mode/ad_block_only_mode_policy_provider.h

```diff
@@ -0,0 +1,52 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_SHIELDS_CORE_COMMON_AD_BLOCK_ONLY_MODE_AD_BLOCK_ONLY_MODE_POLICY_PROVIDER_H_
+#define BRAVE_COMPONENTS_BRAVE_SHIELDS_CORE_COMMON_AD_BLOCK_ONLY_MODE_AD_BLOCK_ONLY_MODE_POLICY_PROVIDER_H_
+
+#include "base/memory/raw_ptr.h"
+#include "base/memory/weak_ptr.h"
+#include "components/prefs/pref_change_registrar.h"
+
+class PrefService;
+
+namespace policy {
+class ConfigurationPolicyProvider;
+class PolicyBundle;
+}  // namespace policy
+
+namespace brave_shields {
+
+class AdBlockOnlyModePolicyProvider {
```

Comment:
per above this doesn't even appear to be a policy provider because it doesn't subclass any type of policy provider.

Reply:
Renamed the class to `AdBlockOnlyModePoliciesLoader` so now it makes more sense now.